### ISSUE TITLE
'Delete' from bulkloader button addition?

### ISSUE THREAD

**Issue by diannakrejsa** [ArctosDB/arctos#2472]
I have a suggestion for a button addition that may help many bulkloading operators on Arctos. Though it is 100% my fault when I forget to delete items from the bulkloader, it typically happens when there are errors to address. I get this screen:
Screen Shot 2020-01-28 at 9.47.23 AM

I select 'download data with errors' and get my spreadsheet to review, then, at times, promptly forget to clear the bulkloader. I have encountered this happening to other Arctos users as well. Maybe, if we can't have multiple bulkloader "lanes," simply having an easily-clicked and readable "delete from bulkloader" among the options on that download data with errors screen would reduce the frequency of forgotten files in the bulkloader? Otherwise one must more intentionally go back through the drop down menus, navigate to the bulkloader, and delete from there.
Screen Shot 2020-01-28 at 9.47.57 AM

Maybe not worth it's time or other hiccups it may cause (and instead I should stop getting distracted so quickly by my error spreadsheet), but a thought!

**Comment by dustymc** [comment]
I presume this is in reference to the bulkloader stage - those screenshots would be useful!

The current single-user design is intentional - everything's an all-or-nothing operation and "enteredby" may be ignored, which greatly simplifies things. It's worked well, but perhaps we have in fact grown past the point where it's a viable solution. The situation the design is intended to avoid is this:

* User A loads records "enteredby" themselves
* User B loads  loads records "enteredby" both themselves and User A
* User B either cannot manage all of the records they just loaded, or they can delete records they didn't enter and which may not be replicated elsewhere.

**Comment by diannakrejsa** [comment]
![1](https://user-images.githubusercontent.com/36166607/73283186-57bccb80-41b8-11ea-9856-3eb6b99ab230.png)

![2](https://user-images.githubusercontent.com/36166607/73283196-5ab7bc00-41b8-11ea-9b85-f00b82451564.png)

Here are the screenshots! I needed to restart so github would be happier with me.

The single lane makes sense then, and generally isn't too much of an issue. I really only use that "lane" for about 15 seconds to load records and check for errors, then whether they're good to bulkload or need review I delete. So the real way the issue comes up is probably people (or, me) just forgetting a file in there. Maybe a clear and present delete button on the first screen (first attachment above) would help?

**Comment by dustymc** [comment]
>forgetting a file 

We could potentially do more with notifications or similar, if we could find a suitable trigger. The load time is hugely variable - a couple records takes a couple seconds to process, large loads may take days - which complicates things a bit.


>clear and present delete button

1. I'm going to find a way to use that terminology...
2. That sort of thing is relatively easy to do (or will be once we get back to a sane development pathway) and always welcome. It could be new button somewhere, or some sort of sticky 'managed staged data' expando-widget on all views of those data.

**Comment by Jegelewicz** [comment]
How about adding the option "download data with errors and delete data from the bulkloader"?

**Comment by diannakrejsa** [comment]
> How about adding the option "download data with errors and delete data from the bulkloader"?

I like that! A one-click stop

**Comment by campmlc** [comment]
I agree with that option. It also makes it more consistent with other bulkloaders, some of which autodelete on load and some which do not. If we are hoping to train people to be consistent and remember a series of complex steps, we should make the steps themselves consistent across tools.

**Comment by dustymc** [comment]
I definitely like consistency, but this isn't a "normal" form - it can do a bunch of things, none of which are precisely "load." It's a multi-function loader-loader.

A big-picture consideration is the "multi-function" aspect of that. I have come to think that most people most of the time are happier with simpler, single-purpose forms (even if they share a back end with other forms), but the cost of that is (perhaps greatly) increased complexity in FINDING the right form, and sometimes clicking a few buttons instead of one.
